relay-server/src/actors/events.rs

```diff
@@ -754,6 +764,31 @@ impl EventProcessor {
         }));
     }
 
+    /// Extracts the timestamp from the minidump and uses it as the event timestamp.
+    #[cfg(feature = "processing")]
+    fn write_minidump_timestamp(event: &mut Event, minidump_item: &envelope::Item) {
+        use std::time::SystemTime;
+
+        assert_eq!(
+            minidump_item.content_type(),
+            Some(&envelope::ContentType::Minidump)
+        );
+        let cursor = std::io::Cursor::new(minidump_item.payload());
+        let minidump = match minidump::Minidump::read(cursor) {
+            Ok(minidump) => minidump,
+            Err(err) => {
+                log::error!("Failed to parse minidump: {:?}", err);
+                metric!(counter(MinidumpParseCounter) += 1, status = "err");
+                return;
+            }
+        };
+        let epoch_offset = Duration::from_secs(minidump.header.time_date_stamp.into());
+        let when: DateTime<Utc> = DateTime::from(SystemTime::UNIX_EPOCH + epoch_offset);
```

Comment:
You can simplify this using [`TimeZone::timestamp`](https://docs.rs/chrono/0.4.13/chrono/offset/trait.TimeZone.html#method.timestamp):

```suggestion
        let timestamp = Utc::timestamp(minidump.header.time_date_stamp.into(), 0);
```